### PR TITLE
Buffer: reimplement UTF encoders with the new UTF Bytes encoders.

### PR DESCRIPTION
**Pull request by dbuenzli**
Note this is not urgent, if there's no time to review it can afford to wait the reheat. 

We just avoid bloating the `Stdlib` with multiple UTF decoding functions by reimplementing the existing `Buffer` UTF encoders with the new `Bytes` encoder that were just merged.

### CODE REVIEW

File: stdlib/buffer.ml
Comment (nojb) [inline stdlib/buffer.ml:146]: Not proposing to change this, but an alternative could be:
```
  resize b (Uchar.utf_8_byte_length u);
  let n = Bytres.set_utf_8_uchar b.buffer pos u in
  b.position <- pos + n
```

**Review by nojb — APPROVED** [review]
LGTM

I left a comment with an alternative implementation; but I don't see clearly which one is better.

File: stdlib/buffer.ml
Comment (dbuenzli) [reply stdlib/buffer.ml:146]: It looks much better but I'd rather not do the case analysis twice in a row on the `u`, i.e. not call `Uchar.utf_8_byte_length`. I could do with replacing it by `uchar_utf_8_byte_length_max` but unless I'm mistaken a call to `resize` will always allocate a new `bytes` value so that would yield miserable performance :–)

The current implementation scheme is modelled after the `add_char` function.

Comment (nojb) [reply stdlib/buffer.ml:146]: Fair enough!

**Comment by nojb** [comment]
@Octachron do you mind reviewing this easy PR, given that you reviewed the other one as well? Thanks!

**Comment by Octachron** [comment]
I should have the time to do a review by the middle of next week.

**Comment by gasche** [comment]
Note: this is not an API change, merely an internal implementation change, so I think a single reviewer would be fine. I'll let @nojb decide if he feels like merging now.

**Comment by nojb** [comment]
> Note: this is not an API change, merely an internal implementation change, so I think a single reviewer would be fine. I'll let @nojb decide if he feels like merging now.

Right, sorry. Anyway, I don't think there is any hurry to merge this as long as it is done before the freeze, so I would let @Octachron have a go at it if he is so inclined :)

**Review by Octachron — APPROVED** [review]
After reviewing the code, this is a welcome and straightforward simplification.

**Comment by nojb** [comment]
@dbuenzli Do you want to add a Changes entry? Otherwise this is good to go.

**Comment by dbuenzli** [comment]
Thanks @Octachron. I don't think a changes entry is needed.

**Comment by nojb** [comment]
Thanks, merged.

**Pull request merged**
